### PR TITLE
Geenie

### PR DESCRIPTION
**Pull request by yannickchoni**
Help me how to return an <item-not-found/> error

### CODE REVIEW

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 1]
```diff
@@ -915,4 +918,18 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
...
+        try {
+            // Delete an unexisting node
+            pubSubManagerOne.deleteNode(nodename);
+
+            assertEquals(null, pubSubManagerOne.getNode(nodename));
+        }
+        catch (PubSubException.NotAPubSubNodeException e){
+            throw new AssertionError("The node can not be deleted",e);
+        }
+    }
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java:949]: For an example of how to assert that an exception occurs, look at https://github.com/goodbytes/smack-sint-extensions/blob/master/src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java#L154-L164

In psuedo-code, you probably want something like this:

```
try 
{
  // do something that should fail.
  doSomethingThatMustFail();

  // fail the test (the code should not reach this line, as the line above should cause the exception to be thrown).
  fail("The server should have returned an error, but did not.");
} 
catch ( /* exception that you expect to happen */ )
{
  // assert that the exception is what you expect (for example: contains a specific condition).
  assert( exception has condition X );
}

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 2]
```diff
@@ -680,7 +680,7 @@ public void unsubscribeInsufficientPrivilegesTest() throws NoResponseException,
      * Asserts that the server returns an 'item-not-found' error response when
      * unsubscribing from a node that does not exist.
      *
-     * <p>From XEP-0060 § 6.2.3.3:</p>
+     * <p>From XEP-0060 § 6.2.3.4:</p>
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java:680]: Good catch! ;)

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 3]
```diff
@@ -919,17 +919,33 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
         }
     }
 
+    /**
+     * Asserts that the server returns an 'item-not-found' error response when
+     * deleting a node that does not exist.
+     *
+     * <p>
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java:923]: I think this reference is incorrect? https://xmpp.org/extensions/xep-0060.html#owner-delete-error-node

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 4]
```diff
@@ -919,17 +919,33 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
...
+
     @SmackIntegrationTest
     public void deleteNonExistentNodeTest() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
         final String nodename = "sinttest-publish-item-nodename-" + testRunId;
         try {
             // Delete an unexisting node
             pubSubManagerOne.deleteNode(nodename);
-
-            assertEquals(null, pubSubManagerOne.getNode(nodename));
+
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java:944]: This test will not fail if the server does not return an error!

File: pom.xml [hunk 5]
```diff
@@ -83,6 +83,11 @@
             <artifactId>junit-jupiter-api</artifactId>
             <version>5.4.2</version>
         </dependency>
+        <dependency>
```
Comment (guusdk) [inline pom.xml:86]: Do we still need this dependency?

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 6]
```diff
@@ -915,4 +915,34 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
...
     }
+
+    /**
+     * Asserts that the server returns an 'item-not-found' error response when
+     * deleting a node that does not exist.
+     *
+     * <p>
+     * From XEP-0060 § 8.4.3.2:
+     * </p>
+     * <blockquote> If the node does not exist, the pubsub service MUST return an
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java]: The javadoc is quoting text from a section of the XEP, but the quoted text is not from the paragraph that is being referred to!

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 7]
```diff
@@ -915,4 +915,36 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
...
+     * @throws NoResponseException   if there was no response from the remote
+     *                               entity.
+     * @throws XMPPErrorException    if there was an XMPP error returned.
+     * @throws NotConnectedException if the XMPP connection is not connected.
+     * @throws InterruptedException  if the calling thread was interrupted.
+     */
+
+    @SmackIntegrationTest
+    public void deleteNonExistentNodeTest() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        final String nodename = "sinttest-publish-item-nodename-" + testRunId;
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java]: It is confusing to give the node a name that refers to a different test. Change the name of the node to `"sinttest-delete-node-that-does-not-exist-" + testRunId` (or something similar).

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java [hunk 8]
```diff
@@ -915,4 +915,36 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
...
+        try {
+            // Delete an unexisting node
```
Comment (guusdk) [inline src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java]: 'unexisting' is not a word.

File: src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java (hunk 7 above)
Comment (yannickchoni) [reply src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java]: It is now changed

File: pom.xml (hunk 5 above)
Comment (yannickchoni) [reply pom.xml:86]: Personaally, I added the smack-legacy dependency, not that one !

Comment (guusdk) [reply pom.xml:86]: smack-legacy is what I mean (the github interface shows only the first line of what I clicked on). Does the code still need smack-legacy?

Comment (yannickchoni) [reply pom.xml:86]: Yes

Comment (yannickchoni) [reply pom.xml:86]: I used that it when I used the MessageEvent class

**Pull request merged**